src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -583,14 +583,25 @@ static int add_partition_esp(DissectedPartition *p, bool has_xbootldr) {
                 if (r < 0)
                         return r;
                 if (r == 0) {
-                        r = path_is_busy("/boot");
-                        if (r < 0)
-                                return r;
-                        if (r == 0) {
+                        k = path_is_busy("/boot");
+                        if (k < 0)
+                                return k;
+                        if (k == 0) {
                                 esp_path = "/boot";
                                 id = "boot";
                         }
                 }
+                if (r > 0) {
```

Comment:
Not a big matter, but why not `} else {` ??
Then, it is not necessary to introduce `k`.